src/status_im2/contexts/wallet/events.cljs

```diff
@@ -311,6 +312,43 @@
                :wallet/valid-ens-or-address?
                false)}))
 
+(rf/reg-event-fx :wallet/find-ens
+ (fn [{:keys [db]} [input contacts chain-id cb]]
+   (let [result (if (empty? input)
+                  []
+                  (filter #(string/starts-with? (or (:ens-name %) "") input) contacts))]
+     (if (and input (empty? result))
+       (rf/dispatch [:wallet/search-ens input chain-id cb ".stateofus.eth"])
+       {:db (assoc db
+                   :wallet/local-suggestions
+                   (map #(assoc % :type item-types/saved-address) result)
+                   :wallet/valid-ens-or-address? (not-empty result))}))))
+
+(rf/reg-event-fx :wallet/search-ens
+ (fn [_ [input chain-id cb domain]]
+   (let [ens (if (string/includes? input ".") input (str input domain))]
+     {:fx [[:json-rpc/call
+            [{:method     "ens_addressOf"
+              :params     [chain-id ens]
+              :on-success #(rf/dispatch [:wallet/set-ens-address % ens])
+              :on-error   (fn []
+                            (if (= domain ".stateofus.eth")
+                              (rf/dispatch [:wallet/search-ens input chain-id cb ".eth"])
```

Comment:
I guess it's best we let the user complete the ENS name with the domain and do the RPC call for fetching the address. We should ask the Desktop team how they handle this.

Additionally, we should move the domains `.stateofus.eth` to a constant.